pkg/clients/aws.go

```diff
@@ -235,6 +236,109 @@ func UsePodServiceAccount(ctx context.Context, _ []byte, _, region string) (*aws
 	return &config, err
 }
 
+// NOTE(muvaf): ACK-generated controllers use aws/aws-sdk-go instead of
+// aws/aws-sdk-go-v2. These functions are implemented to be used by those controllers.
+
+// GetConfigV1 constructs an *awsv1.Config that can be used to authenticate to AWS
+// API by the AWSv1 clients.
+func GetConfigV1(ctx context.Context, c client.Client, mg resource.Managed, region string) (*awsv1.Config, error) {
+	if mg.GetProviderConfigReference() == nil {
+		return nil, errors.New("providerConfigRef cannot be empty")
+	}
+	pc := &v1beta1.ProviderConfig{}
+	if err := c.Get(ctx, types.NamespacedName{Name: mg.GetProviderConfigReference().Name}, pc); err != nil {
+		return nil, errors.Wrap(err, "cannot get referenced Provider")
+	}
+
+	t := resource.NewProviderConfigUsageTracker(c, &v1beta1.ProviderConfigUsage{})
+	if err := t.Track(ctx, mg); err != nil {
+		return nil, errors.Wrap(err, "cannot track ProviderConfig usage")
+	}
+
+	// TODO(muvaf): Support IRSA.
```

Comment:
Looks like you are supporting IRSA already?

Reply:
Ah, good catch!